### PR TITLE
Migrate to Swift 4

### PR DESCRIPTION
**Pull request by juancazalla**
Sorry, this review is huge 😅

### CODE REVIEW

File: ScalaDays/Podfile [hunk 1]
```diff
@@ -6,27 +6,30 @@ platform :ios, '8.0'
...
+    pod 'UIView+AutoLayout', '~> 1.3'
+    
+    # QR code library
+    pod 'ZBarSDK', '~> 1.3.1'
+    
+    # Miscellaneous Utils
+    pod 'NSDate+TimeAgo', '~> 1.0.2'
+    
+    # Push notifications
+    pod 'Localytics', '~> 5.1.0'
```
Comment (juancazalla) [inline ScalaDays/Podfile]: Updated to the last version because of a security warning

File: ScalaDays/Podfile [hunk 2]
```diff
@@ -6,27 +6,30 @@ platform :ios, '8.0'
 inhibit_all_warnings!
 use_frameworks!
 
-target "ScalaDays" do
```
Comment (juancazalla) [inline ScalaDays/Podfile]: Migrated to abstract_target to have Alamofire and SwiftyJSON in tests

File: ScalaDays/ScalaDays/Controllers/SDSocialViewController.swift [hunk 3]
```diff
@@ -17,7 +17,7 @@
 import UIKit
 import SVProgressHUD
 
-class SDSocialViewController: GAITrackedViewController, SDErrorPlaceholderViewDelegate, SDMenuControllerItem {
+class SDSocialViewController: GAITrackedViewController, UITableViewDelegate, UITableViewDataSource, SDErrorPlaceholderViewDelegate, SDMenuControllerItem {
```
Comment (juancazalla) [inline ScalaDays/ScalaDays/Controllers/SDSocialViewController.swift]: Added missing UITableViewDelegate and UITableViewDataSource

File: ScalaDays/ScalaDays/Controllers/SDSponsorViewController.swift [hunk 4]
```diff
@@ -17,7 +17,7 @@
 import UIKit
 import SVProgressHUD
 
-class SDSponsorViewController: GAITrackedViewController, SDErrorPlaceholderViewDelegate, SDMenuControllerItem {
+class SDSponsorViewController: GAITrackedViewController, UITableViewDelegate, UITableViewDataSource, SDErrorPlaceholderViewDelegate, SDMenuControllerItem {
```
Comment (juancazalla) [inline ScalaDays/ScalaDays/Controllers/SDSponsorViewController.swift]: Added missing UITableViewDelegate and UITableViewDataSource

**Comment by juancazalla** [comment]
One question about Crashlytics, can't we add the pod?

**Comment by jdesiloniz** [comment]
@juancazalla Sorry if I take my time with this one... First comment: as you mention I think we could use the Crashlytics pod. If I don't recall wrong, we couldn't use it back in the day, probably because they didn't have a podspec by then! If its pod doesn't work, maybe we could introduce it as a dependency but not including it in the PR (indicating in the README the steps to include the lib and how to compile for external users).

I hope to find more time later to keep with the review, but those were my first two cents on it :D.

**Comment by jdesiloniz** [comment]
@juancazalla I find most of the stuff alright. Could you check if you can get the Crashlytics/Fabric stuff out of the PR (ideally by using Pods)? I think with that sorted out we could think about merging this. Thanks!! :D

File: ScalaDays/Podfile [hunk 5]
```diff
@@ -6,27 +6,33 @@ platform :ios, '8.0'
...
+  # Alamofire
+  pod 'Alamofire', :git => 'https://github.com/Alamofire/Alamofire.git', :tag => '4.5.1'
+  pod 'SwiftyJSON', '3.1.4'
+  
+  target 'ScalaDays' do
+    # Google Analytics
+    pod 'GoogleAnalytics'
+    
+    # Crashlytics
+    pod 'Crashlytics', '3.10.1'
```
Comment (juancazalla) [inline ScalaDays/Podfile]: Now we have Crashlytics

**Comment by juancazalla** [comment]
@jdesiloniz updated removing Crashlytics.framework and adding it to the podfile.

File: ScalaDays/Podfile (hunk 5 above)
Comment (jdesiloniz) [reply ScalaDays/Podfile]: Habemus Crashlytics! 🙌

**Pull request merged**